File: fixtures/clike/file1.kt
package com.example.utils

class DataProcessor(private val maxLength: Int = 255) {

    private val items = mutableListOf<String>()

    fun validateInput(value: String): String {
        require(value.isNotBlank()) { "Input cannot be empty" }
        require(value.length <= maxLength) { "Input exceeds maximum length" }
        return value.trim()
    }

    fun addItem(item: String): DataProcessor {
        items.add(validateInput(item))
        return this
    }

    fun removeItem(item: String): DataProcessor {
        items.remove(item)
        return this
    }

    fun count(): Int = items.size

    fun isEmpty(): Boolean = items.isEmpty()

    fun filterItems(predicate: (String) -> Boolean): List<String> {
        return items.filter(predicate)
    }

    fun <R> mapItems(transform: (String) -> R): List<R> {
        return items.map(transform)
    }

    // File-specific: sum numeric values
    fun sumValues(): Long {
        return items.sumOf { it.toLongOrNull() ?: 0L }
    }
}

fun retryOnFailure(maxAttempts: Int, action: () -> Unit) {
    var attempt = 0
    while (attempt < maxAttempts) {
        try {
            action()
            return
        } catch (e: Exception) {
            attempt++
            if (attempt >= maxAttempts) throw e
            Thread.sleep(1000)
        }
    }
}

File: fixtures/lua/file1.lua
-- Common Lua utility functions

local M = {}

function M.validate_input(value)
    if not value or value == "" then
        return nil, "Input cannot be empty"
    end
    if #value > 255 then
        return nil, "Input exceeds maximum length"
    end
    return value, nil
end

function M.table_contains(tbl, element)
    for _, v in ipairs(tbl) do
        if v == element then return true end
    end
    return false
end

function M.map(tbl, fn)
    local result = {}
    for i, v in ipairs(tbl) do
        result[i] = fn(v)
    end
    return result
end

function M.filter(tbl, predicate)
    local result = {}
    for _, v in ipairs(tbl) do
        if predicate(v) then
            result[#result + 1] = v
        end
    end
    return result
end

function M.reduce(tbl, fn, init)
    local acc = init
    for _, v in ipairs(tbl) do
        acc = fn(acc, v)
    end
    return acc
end

-- File-specific: sum a table
function M.sum(tbl)
    return M.reduce(tbl, function(a, b) return a + b end, 0)
end

return M
